pytorch_lightning/loggers/wandb.py

```diff
@@ -213,8 +233,42 @@ def version(self) -> Optional[str]:
         # don't create an experiment if we don't have one
         return self._experiment.id if self._experiment else self._id
 
+    @rank_zero_only
+    def after_save_checkpoint(self, checkpoint_callback: 'ReferenceType[ModelCheckpoint]') -> None:
+        # log checkpoints as artifacts
+        if self._log_model == 'all' or self._log_model is True and checkpoint_callback.save_top_k == -1:
+            self._scan_and_log_checkpoints(checkpoint_callback)
+        elif self._log_model is True:
+            self._checkpoint_callback = checkpoint_callback
+
     @rank_zero_only
     def finalize(self, status: str) -> None:
-        # upload all checkpoints from saving dir
-        if self._log_model:
-            wandb.save(os.path.join(self.save_dir, "*.ckpt"))
+        # log checkpoints as artifacts
+        if self._checkpoint_callback:
+            self._scan_and_log_checkpoints(self._checkpoint_callback)
+
+    def _scan_and_log_checkpoints(self, checkpoint_callback: 'ReferenceType[ModelCheckpoint]') -> None:
+        # get checkpoints to be saved with associated score
+        checkpoints = {
+            checkpoint_callback.last_model_path: checkpoint_callback.current_score,
+            checkpoint_callback.best_model_path: checkpoint_callback.best_model_score,
+            **checkpoint_callback.best_k_models}
+        checkpoints = sorted([(Path(p).stat().st_mtime, p, s)
```

Comment:
Hey, 2 small questions:
1. Do you see a need to use the time to filer previous checkpoint. Won't it be simpler to keep a list of upload checkpoint.
2. What happens if the training fails and the user re-starts from a given checkpoint.  `_logged_model_time` will be empty and all checkpoints will be re-uploaded.

Reply:
1. The issue is you could have checkpoints with the same name (eg: `last.ckpt`) so the time helps to know it's a new file.
2. If the training fails, I imagine you would start a new run (maybe you would even reload from a previous checkpoint), so you would want to upload the checkpoints. Each run is associated to a list of checkpoints contained within an artifact name. 

When you want to reload a model, you would choose the run and then the version of your artifact (typically the one we will call `best` with this integration). I'm currently writing [a demo](https://colab.research.google.com/drive/1Tr9dYlwBKk6-LgLKGO8KYZULnguVA992#scrollTo=4rFC9ICWazTy) so you can see what it does